streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -969,7 +969,7 @@ boolean sendingOldValueEnabled() {
                     foreignKeyExtractor,
                     foreignKeySerde,
                     subscriptionTopicName,
-                    valSerde.serializer(),
+                    valSerde == null ? null : valSerde.serializer(),
```

Comment:
In addition to the "wrapping serializer" change, there were several places where we would directly throw an NPE while attempting to get the serializer from the current DSL operator's nullable `valSerde`.